src/main/java/com/amihaiemil/docker/RtContainer.java

```diff
@@ -38,7 +44,32 @@
 
     @Override
     public JsonObject inspect() throws IOException {
-        return null;
+        final HttpGet inspect = new HttpGet(this.baseUri.toString() + "/json");
+        final HttpResponse response = this.client.execute(inspect);
+        final JsonObject info;
+        if(response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+            info = Json
+                .createReader(response.getEntity().getContent()).readObject();
+        } else {
+            info = null;
+        }
+        inspect.releaseConnection();
```

Comment:
@amihaiemil see [this](https://stackoverflow.com/a/31659073/1623885). I think we should leave a puzzle for abstracting away all the low-level apache http handling stuff into a single point. Something like "Encapsulate HTTP connection/request/response handling" I guess. We'd gain:

* Reduced risk per PR since we won't have to keep implementing/reviewing correct handling of HTTP resources
* Reduce procedural code throughout the code base